### ISSUE TITLE
Investigate AND verify stuck & missing donations

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/impact-graph#1556]
On thursday/friday a user reported their donations were not showing up on ETH mainnet to this project 

https://giveth.io/project/emergency-relief-fund-for-brazil-floods

I was able to confirm this by making a donation myself in ETH to their project on mainnet, I cannot find the donation in 'my donations' or on the project table, nor in the back-up service. 

Here are the three donations we know of that we're not caught anywhere: 
https://etherscan.io/tx/0x331ff224c7e4dcc3ce9bb6aa6b3a67e1d7b1c7f1c27e6fdead150fd09a6499b6
https://etherscan.io/tx/0xf1a4fb4c446c25529fed988c83381db41fd6cbbe8aae69a58f2a93a60a8b9f73
https://etherscan.io/tx/0x0373f7efaeb12c205e2c26f2b14a49194b9e619c6728c8a5a4f8cf36f635ef6f

Alternatively, when I was testing donations to other projects on ETH mainnet my donations show up in my donations but they are stuck in a pending state, I also cannot find them in the back-up service. I made some donations on Optimism mainnet and encountered no issues for comarison.

![image](https://github.com/Giveth/impact-graph/assets/67759413/def66460-8aaf-4ff5-82f5-eb78669994b8)

Here are the txs for my stuck pending donations:
https://etherscan.io/tx/0xc81021a481b1a21947b1aed7199cd7febdd662cd9a4ee839eb734849b59b24c2
https://etherscan.io/tx/0x5c4cab8d517393178a6f59af693185848c57b21975710cb71f706e8984b117db

--- 

I tested today, late friday making a donation on eth mainnet this donation was successful and show on my donations and the projects donations table. however the previous donations on eth mainnet do not show. 

How did this happen? We should 1st add the missing donations into our DB. 2nd we need to investigate how this happens that donations are stuck as pending and how it can occuer that other donations are just not caught by any service at all, but only for eth mainnet

**Comment by divine-comedian** [comment]
Also, there is 61 failed donations in the back-up service that needed to be checked they have been added back to our UI! This is from last week when the donation verification service went down unexpectedly.

![image](https://github.com/Giveth/impact-graph/assets/67759413/7fb2dc35-27ff-4e2c-a404-aa913ae62f02)

**Comment by mohammadranjbarz** [comment]
@divine-comedian 
I checked the DB for your first three transactions (https://etherscan.io/tx/0x331ff224c7e4dcc3ce9bb6aa6b3a67e1d7b1c7f1c27e6fdead150fd09a6499b6, https://etherscan.io/tx/0xf1a4fb4c446c25529fed988c83381db41fd6cbbe8aae69a58f2a93a60a8b9f73,
https://etherscan.io/tx/0x0373f7efaeb12c205e2c26f2b14a49194b9e619c6728c8a5a4f8cf36f635ef6f)
We have them in the DB in draft_donation table, and they are not matched(Still pending) so this is the problem that is why we haven't matched them although they have been successful.

And about the other two TXs that @divine-comedian mentioned, now we have the TXs in our db for those, so there isn't any problem related to them
https://etherscan.io/tx/0xc81021a481b1a21947b1aed7199cd7febdd662cd9a4ee839eb734849b59b24c2
https://etherscan.io/tx/0x5c4cab8d517393178a6f59af693185848c57b21975710cb71f706e8984b117db

**TL;DR**
Unfortunately, our draft donation service doesn't work properly.

I will post more details as I find any reason

**Comment by geleeroyale** [comment]
![image](https://github.com/Giveth/impact-graph/assets/317685/156f317a-25d9-49c4-a1d5-304140acc656)

Thanks @mohammadranjbarz - unfortunately I was not able to replicate the issue @divine-comedian was facing (my donation showed instantly and as complete) which leads me to concur with @mohammadranjbarz that we need to take a look at backup service. 

I was investigating a lead regarding outdated infura endpoints (which I would still like to revisit, but need someone from **frontend** to take a look) but seeing it works now I am assuming the problem was temporary. I did check for incidents on infuras side for the relevant time, but nothing out of the ordinary.

**Comment by divine-comedian** [comment]
So far from user reports we're missing the following donations: 

#### https://giveth.io/project/emergency-relief-fund-for-brazil-floods
- [x] https://etherscan.io/tx/0x0373f7efaeb12c205e2c26f2b14a49194b9e619c6728c8a5a4f8cf36f635ef6f
- [x] https://etherscan.io/tx/0xf1a4fb4c446c25529fed988c83381db41fd6cbbe8aae69a58f2a93a60a8b9f73

#### https://giveth.io/es/project/skatehive-skateboarding-community?tab=donations
- [x] https://etherscan.io/tx/0x75558bed003bb9b55c4ceaddce6119f228db8ec217bb90a414afd0bc1cb97965
- [x] https://gnosisscan.io/tx/0xfa65ef0a52e2f3b96c5802dcee4783858511989b7235035e8cab4d527fa15a1a

**Comment by mohammadranjbarz** [comment]
> So far from user reports we're missing the following donations:
> 
> #### https://giveth.io/project/emergency-relief-fund-for-brazil-floods
> * [ ]  https://etherscan.io/tx/0x0373f7efaeb12c205e2c26f2b14a49194b9e619c6728c8a5a4f8cf36f635ef6f
> * [ ]  https://etherscan.io/tx/0xf1a4fb4c446c25529fed988c83381db41fd6cbbe8aae69a58f2a93a60a8b9f73
> 
> #### https://giveth.io/es/project/skatehive-skateboarding-community?tab=donations
> * [ ]  https://etherscan.io/tx/0x75558bed003bb9b55c4ceaddce6119f228db8ec217bb90a414afd0bc1cb97965
> * [ ]  https://gnosisscan.io/tx/0xfa65ef0a52e2f3b96c5802dcee4783858511989b7235035e8cab4d527fa15a1a

@divine-comedian Now about 70 donations have been created by draft donation service and got verified, four TXs that you already said 3 of them have been added but the fourth one has not matched yet, so this problem is almost fixed, but not 100%,  I'm still working

**Comment by laurenluz** [comment]
@mohammadranjbarz when you merge this today, all the missed donations should be considered, right? wondering because GIvbacks and QF data analysis are blocked 😅

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz when you merge this today, all the missed donations should be considered, right? wondering because GIvbacks and QF data analysis are blocked 😅

Yess,  I just need to check the recurring donations streams, it would work but I need to check the documents in the DB

**Comment by laurenluz** [comment]
what's the status on this?

**Comment by jainkrati** [comment]
@maryjaf to confirm if this is working on staging.

**Comment by maryjaf** [comment]
> @maryjaf to confirm if this is working on staging.

my missing donation to https://giveth.io/project/commons-stackproject on 6 May,  have been added correctly 
but for other donations for any users, I don't know how can I check @mohammadranjbarz

**Comment by divine-comedian** [comment]
There is just one transaction I referenced here that needs to be added manually since it didn't appear caught in any service
#1580

**Comment by jainkrati** [comment]
@CarlosQ96 can you pls take this up as @mohammadranjbarz is not well.

**Comment by divine-comedian** [comment]
Mohammad added in manually the one missing transaction, we can close this one.
